.github/workflows/codeql.yml

```diff
@@ -0,0 +1,69 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ['java', 'go']
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          # If you wish to specify custom queries, you can do so here or in a config file.
+          # By default, queries listed here will override any specified in a config file.
+          # Prefix the list here with "+" to use these queries and those in the config file.
+          # queries: ./path/to/local/query, your-org/your-repo/queries@main
+          languages: ${{ matrix.language }}
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        if: matrix.language == 'java'
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 11
+        if: matrix.language == 'java'
+
+      # https://docs.gradle.org/current/userguide/performance.html
+      - name: Build
+        run: ./gradlew clean assemble compileTestJava --no-build-cache --parallel --daemon
```

Comment:
I am not familiar with your usage of CodeQL, so I made the assumption that you used it to scan test code as well. If you do not, the `compileTestJava` task can be removed here.

Reply:
Based on our previous experience, we would like to execute the `build` task. The `build` task should already include the `assemble` and `compileTestJava` tasks. In your experience, do you think it's necessary to replace the `build` task with the `assemble` and `compileTestJava` tasks?

Reply:
`build` will run everything in the build, most notably the tests. The tests will be verified in the Continuous Integration workflow, so running them here would be redundant (unless I misunderstand and CodeQL needs to see them execute). Running `assemble compileTestJava` will just compile main and test source set that CodeQL cares about.

If you find it better, we could change this to something like `build -x test` for a similar result.

Reply:
OK, let's keep `assemble compileTestJava` then. This way, only the source code will be compiled, but CodeQL doesn't need them to run.